solver/src/main/java/DbConstraintSolverZ3InDocker.java

```diff
@@ -96,54 +104,137 @@ String solveFromFile(String fileName) {
-                    null,
-                    null,
-                    false,
-                    false);
-            genes.put(name, gene);
+
+            solved.put(values[0], values[1]);
         }
+        return solved;
+    }
 
-        // TODO: Handle more than 1 table
-        String tableName = this.schemaDto.tables.get(0).name;
-        List<String> history = new LinkedList<>();
-        Set<String> columnNames = new HashSet<>(Collections.singletonList("*"));
+    /**
+     * Given the solved constraints, it returns a list of SqlAction with the necessary inserts according to the constraints
+     * The SqlAction are generated with the SqlInsertBuilder, and then the values are overwritten with the ones obtained with the Z3 execution
+     * @param writer used to get the variable genes the same format that the Z3 solver uses (a prefix for the table and the gene name)
+     * @param solvedConstraints a map with the variables and their values
+     * @return a list of Sql with the necessary inserts according to the constraints
+     */
+    private List<SqlAction> toSqlAction(Smt2Writer writer, Map<String, String> solvedConstraints) {
 
-        List<SqlAction> actions = sqlInsertBuilder.createSqlInsertionAction(tableName,
-                columnNames, history, false, false, false);
+        List<SqlAction> actions = new LinkedList<>();
+        for (TableDto table : this.schemaDto.tables) {
 
-        actions.get(0).seeTopGenes().forEach(g -> {
-            if (genes.containsKey(g.getName())) {
-                g.copyValueFrom(genes.get(g.getName()));
-            } else {
-                log.warn("Gene not found in the model: " + g.getName());
-            }
-        });
+            String tableName = table.name;
+            List<String> history = new LinkedList<>();
+            Set<String> columnNames = new HashSet<>(Collections.singletonList("*"));
+
+            List<SqlAction> newActions = sqlInsertBuilder.createSqlInsertionAction(tableName,
+                    columnNames, history, false, false, false);
+
+            newActions.forEach(action -> action.seeTopGenes().forEach(currentGene -> {
+                String tableVariableKey = writer.asTableVariableKey(table, currentGene.getName());
+                if (solvedConstraints.containsKey(tableVariableKey)) {
+
+                    if (currentGene instanceof NullableGene) {
+                        currentGene = ((NullableGene) currentGene).getGene();
+                    }
+
+                    String solvedValue = solvedConstraints.get(tableVariableKey);
+
+                    if (currentGene instanceof EnumGene) {
+                        // There is nothing to do, as the enum has the values already parsed
+                        return;
+                    }
+
+                    if (currentGene instanceof IntegerGene) {
+                        Integer value =  Integer.parseInt(solvedValue);
+                        Gene geneWithSolvedValue = new IntegerGene(
```

Comment:
@agusaldasoro also, thinking about it... but why creating a new gene here? can't you just modify `value` directly?

Reply:
Yes! done!